apps/web/src/actions/datasetsV2/generateDataset.ts

```diff
@@ -0,0 +1,111 @@
+'use server'
+
+import {
+  ChainStepResponse,
+  CLOUD_MESSAGES,
+  Dataset,
+  LogSources,
+  StreamEventTypes,
+} from '@latitude-data/core/browser'
+import { BadRequestError } from '@latitude-data/core/lib/errors'
+import { createDataset } from '@latitude-data/core/services/datasets/create'
+import { env } from '@latitude-data/env'
+import { ChainEventDto } from '@latitude-data/sdk'
+import slugify from '@sindresorhus/slugify'
+import { createSdk } from '$/app/(private)/_lib/createSdk'
+import { getCurrentUserOrError } from '$/services/auth/getCurrentUser'
+import { createStreamableValue } from 'ai/rsc'
+
+type GenerateDatasetActionProps = {
+  parameters: Record<string, unknown>
+  description: string
+  rowCount: number
+  name: string
+}
+
+export async function generateDatasetAction({
```

Comment:
This does not need to be streaming. Use a regular action.

Redirect to detail when successfully created